instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/handlers/default.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActiveJob
+      module Handlers
+        # Default handler to create internal spans for events
+        # This class provides default template methods that derived classes may override to generate spans and register contexts.
+        class Default
+          # @param parent_span_provider [Object] provides access to the top most parent span (usually the ingress span)
+          # @param mapper [Callable] converts ActiveSupport::Notifications payloads to span attributes
+          # @param config [Hash] of instrumentation options
+          def initialize(parent_span_provider, mapper, config)
+            @mapper = mapper
+            @config = config
+            @parent_span_provider = parent_span_provider
+          end
+
+          # Invoked by ActiveSupport::Notifications at the start of the instrumentation block
+          # It amends the otel context of a Span and Context tokens to the payload
+          #
+          # @param name [String] of the Event
+          # @param id [String] of the event
+          # @param payload [Hash] containing job run information
+          # @return [Hash] the payload passed as a method argument
+          def start(name, id, payload)
+            payload.merge!(__otel: start_span(name, id, payload))
+          rescue StandardError => e
+            OpenTelemetry.handle_error(exception: e)
+          end
+
+          # Creates a span and registers it with the current context
+          #
+          # @param name [String] of the Event
+          # @param id [String] of the event
+          # @param payload [Hash] containing job run information
+          # @return [Hash] with the span and generated context tokens
+          def start_span(name, _id, payload)
+            span = tracer.start_span(name, attributes: @mapper.call(payload))
+            tokens = [OpenTelemetry::Context.attach(OpenTelemetry::Trace.context_with_span(span))]
+
+            { span: span, ctx_tokens: tokens }
+          end
+
+          # Creates a span and registers it with the current context
+          #
+          # @param _name [String] of the Event (unused)
+          # @param _id [String] of the event (unused)
+          # @param payload [Hash] containing job run information
+          # @return [Hash] with the span and generated context tokens
+          def finish(_name, _id, payload)
+            otel = payload.delete(:__otel)
+            span = otel&.fetch(:span)
+            tokens = otel&.fetch(:ctx_tokens)
+
+            exception = payload[:error] || payload[:exception_object]
+            on_exception(exception, span) if exception
```

Comment:
Reduce memory allocations suggestion, part 1:
```suggestion
            on_exception((payload[:error] || payload[:exception_object]), span)
```